src/useq/_plate.py

```diff
@@ -0,0 +1,429 @@
+        return self.all_well_indices[self.selected_wells].reshape(-1, 2)
+
+    @cached_property
+    def all_well_coordinates(self) -> np.ndarray:
+        """Return the stage coordinates of all wells as array with shape (N, 2)."""
+        return self._transorm_coords(self.all_well_indices.reshape(-1, 2))
+
+    @cached_property
+    def selected_well_coordinates(self) -> np.ndarray:
+        """Return the stage coordinates of selected wells as array with shape (N, 2)."""
+        return self._transorm_coords(self.selected_well_indices)
+
+    @property
+    def all_well_names(self) -> np.ndarray:
+        """Return the names of all wells as array of strings with shape (Rows, Cols)."""
+        return np.array(
+            [
+                [f"{_index_to_row_name(r)}{c+1}" for c in range(self.plate.columns)]
+                for r in range(self.plate.rows)
+            ]
+        )
+
+    @property
+    def selected_well_names(self) -> list[str]:
+        """Return the names of selected wells."""
+        return list(self.all_well_names[self.selected_wells].reshape(-1))
+
+    def _transorm_coords(self, coords: np.ndarray) -> np.ndarray:
+        """Transform coordinates to the plate coordinate system."""
+        # create homogenous coordinates
+        h_coords = np.column_stack((coords, np.ones(coords.shape[0])))
+        # transform
+        transformed = self.affine_transform @ h_coords.T
+        # strip homogenous coordinate
+        return (transformed[:2].T).reshape(coords.shape)  # type: ignore[no-any-return]
+
+    @property
+    def all_well_positions(self) -> Sequence[Position]:
+        """Return all wells (centers) as Position objects."""
+        return [
+            Position(x=x, y=y, name=name)
+            for (y, x), name in zip(
+                self.all_well_coordinates, self.all_well_names.reshape(-1)
+            )
+        ]
+
+    @cached_property
+    def selected_well_positions(self) -> Sequence[Position]:
+        """Return selected wells (centers) as Position objects."""
+        return [
+            Position(x=x, y=y, name=name)
+            for (y, x), name in zip(
+                self.selected_well_coordinates, self.selected_well_names
+            )
+        ]
+
+    @cached_property
+    def image_positions(self) -> Sequence[Position]:
+        """All image positions.
+
+        This includes *both* selected wells and the image positions within each well
+        based on the `well_points_plan`.  This is the primary property that gets used
+        when iterating over the plan.
+        """
+        wpp = self.well_points_plan
+        offsets: Iterable[Position | GridPosition] = (
+            [wpp] if isinstance(wpp, Position) else wpp
+        )
+        # TODO: note that all positions within the same well will currently have the
+        # same name.  This could be improved by modifying `Position.__add__` and
+        # adding a `name` to GridPosition.
```

Comment:
Yes I think this makes sense. We can add a `name` to the `GridPosition` which can be by default a `int` created in the `__iter__` method of the `RandomPoints` and the `iter_grid_positions` of `_GridPlan`. So then we can have A1_0000, A1_0001, etc...

Reply:
`Position` `__add__` could look something like this:

```py
    def __add__(self, other: "Position | GridPosition") -> "Position":
        """Add two positions together to create a new position."""
        if isinstance(other, GridPosition) and not other.is_relative:
            raise ValueError("Cannot add a non-relative GridPosition to a Position")

        other_name = getattr(other, "name", "")
        if other_name:
            if other_name.isdigit():
                other_name = str(other_name).zfill(4)
            other_name = f"_{other_name}"

        other_z = getattr(other, "z", None)

        return Position(
            x=self.x + other.x if self.x is not None and other.x is not None else None,
            y=self.y + other.y if self.y is not None and other.y is not None else None,
            z=self.z + other_z if self.z is not None and other_z is not None else None,
            name=f"{self.name}{other_name}" if self.name and other_name else self.name,
            sequence=self.sequence,
        )
```

`GridPosition` will have the `name` attribute:
```py
class GridPosition(NamedTuple):
    x: float
    y: float
    row: int
    col: int
    is_relative: bool
    name: str
```


the `iter_grid_positions` method of `_GridPlan` could be like this:
```py
    def iter_grid_positions(
        self,
        fov_width: float | None = None,
        fov_height: float | None = None,
        *,
        mode: OrderMode | None = None,
    ) -> Iterator[GridPosition]:
        """Iterate over all grid positions, given a field of view size."""
        _fov_width = fov_width or self.fov_width or 1.0
        _fov_height = fov_height or self.fov_height or 1.0
        mode = self.mode if mode is None else OrderMode(mode)

        dx, dy = self._step_size(_fov_width, _fov_height)
        rows = self._nrows(dy)
        cols = self._ncolumns(dx)
        x0 = self._offset_x(dx)
        y0 = self._offset_y(dy)

        for idx, (r, c) in enumerate(_INDEX_GENERATORS[mode](rows, cols)):
            yield GridPosition(
                x0 + c * dx, y0 - r * dy, r, c, self.is_relative, f"{idx}"
            )
```

and `__iter__` form `RandomPoints`:

```py
    def __iter__(self) -> Iterator[GridPosition]:  # type: ignore
        seed = np.random.RandomState(self.random_seed)
        func = _POINTS_GENERATORS[self.shape]
        n_points = max(self.num_points, MIN_RANDOM_POINTS)
        points: list[Tuple[float, float]] = []
        for idx, (x, y) in enumerate(
            func(seed, n_points, self.max_width, self.max_height)
        ):
            if (
                self.allow_overlap
                or self.fov_width is None
                or self.fov_height is None
                or _is_a_valid_point(points, x, y, self.fov_width, self.fov_height)
            ):
                yield GridPosition(x, y, 0, 0, True, f"{idx}")
                points.append((x, y))
            if len(points) >= self.num_points:
                break
        else:
            warnings.warn(
                f"Unable to generate {self.num_points} non-overlapping points. "
                f"Only {len(points)} points were found.",
                stacklevel=2,
            )
```

so, when we iter through the `WellPlatePlan` we can have:
```
x=99.97823292699483 y=197.6533528400231 name='A1_0000'
x=100.89011127568732 y=201.85455673194744 name='A1_0001'
x=98.65280352634389 y=201.43210273330388 name='A1_0002'
x=108.97823292699483 y=197.6533528400231 name='A2_0000'
x=109.89011127568732 y=201.85455673194744 name='A2_0001'
x=107.65280352634389 y=201.43210273330388 name='A2_0002'
```

Reply:
i'm fine with the outcome (`A1_0001`) but having the zfill inside of `__add__` mixes concerns of something it shouldn't care about.  If grid plan `__iter__`  wants to zfill its own name, that's fine, but all add should do is add the names together or not